javascript/javascript1/week4/get-reply.js

```diff
@@ -0,0 +1,138 @@
+// voice assistant function in javascript
+/* name should be global variable beacause the name is extracted 
+in the first call to getReply and it is displayed in the second call 
+ */
+let name;
+let todo = [];
+function getReply(inputString) {
+  inputString = inputString.trim();
+  if (inputString.search("Hello my name is") == 0) {
+    // this block of code is say hello to user and store user name in the global variable
+    let positionOfName =
+      inputString.search("Hello my name is Benjamin") +
+      "Hello my name is".length;
+    name = inputString.substring(positionOfName);
+    return "Nice to meet you " + name;
+  }
+  if (inputString.search("What is my name") == 0) {
+    // this block of code is dispaly name
+    if (name != null) return "Your name is " + name;
+    else return "Name is not defined";
+  }
+  // the following code is to add activity to "todo list"
+  if (inputString.search("Add") == 0) {
+    // now we have to find the activity
+    let activity = inputString.substring(4, inputString.search("to my todo"));
+    // length of add is 3 and one space so total 4
+    todo.push(activity);
+    return activity + "added to your todo";
+  }
+
+  // the following code is to remove from "todo list"
+  if (inputString.search("Remove") == 0) {
+    let activity = inputString.substring(7, inputString.search("from my todo"));
+    todo.pop(activity);
+    return "Removed " + activity + "from your todo";
+  }
+
+  // the following code is to dispaly the activities todo
+  if (inputString === "What is on my todo?") {
+    let activities = "";
+    for (let i = 0; i < todo.length; i++) {
+      if (i == todo.length - 1) {
+        // this part is to display last activity with and
+        activities += "and " + todo[i];
+      } else {
+        // this part is to display all activities except last with ,
+        activities += todo[i] + ",";
+      }
+    }
+    return activities;
+  }
+
+  // the following code is to dispaly date in human readable format
+  if (inputString === "What day is it today?") {
+    let today = new Date();
+    const monthNames = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    return (
+      today.getDay() +
+      ". of " +
+      monthNames[today.getMonth()] +
+      " " +
+      today.getFullYear()
+    );
+  }
```

Comment:
There is a new method for `Date` at the new JavaScript version to converts a date to a string by using the current or specified locale which is `toLocaleDateString`, you can pass some options to specify the format you want. Example:
```js
  if (inputString === 'what day is it today?') {
    return new Date().toLocaleDateString('en-US', {
      day: 'numeric',
      year: 'numeric',
      month: 'long'
    });
  }
```